tests/vnc_test.go

```diff
@@ -110,5 +117,67 @@ var _ = Describe("VNC", func() {
 				Expect(err).To(BeNil())
 			})
 		})
+
+		table.DescribeTable("should upgrade subresource connections if an origin header is given", func(subresource string) {
+			config, err := kubecli.GetKubevirtClientConfig()
+			Expect(err).ToNot(HaveOccurred())
+			rt, err := upgradeCheckRoundTripperFromConfig(config)
+			Expect(err).ToNot(HaveOccurred())
+			req, err := kubecli.RequestFromConfig(config, vmi.Name, vmi.Namespace, subresource)
+
+			// Add an Origin header to look more like an arbitrary browser
+			if req.Header == nil {
+				req.Header = http.Header{}
+			}
+			req.Header.Add("Origin", config.Host)
+			_, err = rt.RoundTrip(req)
+			Expect(err).ToNot(HaveOccurred())
+		},
+			table.Entry("for vnc", "vnc"),
+			table.Entry("for serial console", "console"),
+		)
 	})
 })
+
+// checkUpgradeRoundTripper checks if an upgrade confirmation is received from the server
+type checkUpgradeRoundTripper struct {
+	Dialer *websocket.Dialer
+}
+
+func (t checkUpgradeRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	GinkgoRecover()
+	conn, resp, err := t.Dialer.Dial(r.URL.String(), r.Header)
+	if err != nil {
+		return nil, err
+	}
+	Expect(resp.StatusCode).To(Equal(http.StatusSwitchingProtocols))
+
+	defer conn.Close()
```

Comment:
I think it must be before `Expect(resp.StatusCode).To(Equal(http.StatusSwitchingProtocols))`